bundles/core/org.eclipse.smarthome.core.ephemeris/src/main/java/org/eclipse/smarthome/core/ephemeris/internal/EphemerisManagerImpl.java

```diff
@@ -0,0 +1,242 @@
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.core.ephemeris.internal;
+
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URL;
+import java.time.DayOfWeek;
+import java.time.LocalDate;
+import java.time.ZonedDateTime;
+import java.time.format.TextStyle;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.config.core.ConfigOptionProvider;
+import org.eclipse.smarthome.config.core.ConfigurableService;
+import org.eclipse.smarthome.config.core.ParameterOption;
+import org.eclipse.smarthome.core.ephemeris.EphemerisManager;
+import org.eclipse.smarthome.core.i18n.LocaleProvider;
+import org.osgi.framework.Constants;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Modified;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import de.jollyday.Holiday;
+import de.jollyday.HolidayManager;
+import de.jollyday.ManagerParameter;
+import de.jollyday.ManagerParameters;
+
+/**
+ * This service provides functionality around ephemeris services and is the central service to be used directly by
+ * others.
+ *
+ * @author Gaël L'hopital - Initial contribution and API
+ */
+@Component(immediate = true, configurationPid = "org.eclipse.smarthome.ephemeris", property = {
+        Constants.SERVICE_PID + "=org.eclipse.smarthome.ephemeris",
+        ConfigurableService.SERVICE_PROPERTY_CATEGORY + "=system",
+        ConfigurableService.SERVICE_PROPERTY_LABEL + "=Ephemeris",
+        ConfigurableService.SERVICE_PROPERTY_DESCRIPTION_URI + "=" + EphemerisManagerImpl.CONFIG_URI })
+public class EphemerisManagerImpl implements EphemerisManager, ConfigOptionProvider {
+    private final Logger logger = LoggerFactory.getLogger(EphemerisManagerImpl.class);
+
+    // constants for the configuration properties
+    protected static final String CONFIG_URI = "system:ephemeris";
+    private static final String CONFIG_DAYSET_PREFIX = "dayset-";
+    private static final String CONFIG_DAYSET_WEEKEND = "weekend";
+    private static final String CONFIG_COUNTRY = "country";
+    private static final String CONFIG_REGION = "region";
+    private static final String CONFIG_CITY = "city";
+
+    private final Map<String, Set<DayOfWeek>> daysets = new HashMap<>();
+    private final Map<Object, HolidayManager> holidayManagers = new HashMap<>();
+
+    private @Nullable String country;
+    private final List<String> countryParameters = new ArrayList<>();
+
+    private @Nullable LocaleProvider localeProvider;
```

Comment:
For mandatory references we don't use `@Nullable` but disable the nullness checking using `@NonNullByDefault({})`